internal/vds/metadatahandle.hpp

```diff
@@ -23,6 +26,43 @@ class MetadataHandle {
 
     Axis const& get_axis(Direction const direction) const noexcept (false);
 
+    /**
+     * Extends IJKCoordinateTransformer with ability to translate to openvds
+     * positions - aka system used in Request- functions.
+
+     * We have a problem that OpenVDS' ToVoxelPosition and data request
+     * functions have different definitions of where a datapoint is. E.g. A
+     * transformer will return (0,0,0) for the first sample in the cube. The
+     * request functions on the other hand assume the data is located in the
+     * center of a voxel. I.e. that the first sample is at (0.5, 0.5, 0.5).
+     *
+     * Thus extending IJKCoordinateTransformer with new methods helps us to
+     * bridge this gap.
+     */
+    class CoordinateTransformer : public OpenVDS::IJKCoordinateTransformer{
+    public:
+        CoordinateTransformer(const MetadataHandle& metadata ) :
+            OpenVDS::IJKCoordinateTransformer(metadata.m_layout), m_metadata(metadata) {};
+
+        OpenvdsVoxelVector IndexToOpenvdsPosition     (const OpenVDS::DoubleVector3& position) const;
+        OpenvdsVoxelVector AnnotationToOpenvdsPosition(const OpenVDS::DoubleVector3& position) const;
+        OpenvdsVoxelVector WorldToOpenvdsPosition     (const OpenVDS::DoubleVector3& position) const;
+
+        OpenvdsVoxelVector ToOpenvdsPosition(
+            const OpenVDS::DoubleVector3& coordinate,
+            enum coordinate_system coordinate_system
+        ) const;
+
+        bool IsOpenvdsPositionOutOfRangeIline (double value) const noexcept(true);
+        bool IsOpenvdsPositionOutOfRangeXline (double value) const noexcept(true);
+        bool IsOpenvdsPositionOutOfRangeSample(double value) const noexcept(true);
+        bool IsOpenvdsPositionOutOfRange   (const OpenvdsVoxelVector& position) const noexcept(true);
+        Axis WhereOpenvdsPositionOutOfRange(const OpenvdsVoxelVector& position) const noexcept(false);
```

Comment:
>Wanted to have everything that extends some openvds functionality to have this prefix to make it obvious that we extend something that shouldn't have been extended, this is not our class.

A noble attempt, but you can't make this definition without documenting it. If that naming convention is supposed to add something to all devs it must be known to all devs. 

Personally I don't see the point in trying to distinguish stuff from openvds with names. Everything in openvds lives under a namespace anyway.

My complaints are made, do with it what you want 😄

Reply:
My initial thought was to just override the methods where we wanted to change the behaviour (Because I'm considering the openvds transformer to be wrong). So conceptually something like this:

```
struct Transformer : private OpenVDS::IJKCoordinateTransformer {
    using base_type = OpenVDS::IJKCoordinateTransformer;
     
    // Expose all the methods we don't want to change
    using base_type::base_type; // constructor

    using base_type::IJKPositionToAnnotation;
    using base_type::IJKIndexToAnnotation;
    ....

    // Custom implementation where needed
     DoubleVector3 WorldToVoxelPosition(DoubleVector3 const& world) const noexcept (false) {
         return base_type::WorldToVoxelPosition(world) + 0.5;
    } 
}
```

This keeps the original interface and avoid the problem of coming up with new names. Feel free to disregard if you don't like it